### DISCUSSION TITLE
Can a server attend more than 1 client?

### DISCUSSION THREAD

**Discussion by augustodarruda** [r-simmer/simmer#270] [Q&A] [answered]
Is there a way that a server attend more than 1 client? 

Since I'm modelling a live chat call center, the operation allows a server to attend 4 clients at once, when a client leaves, another client that was on queue joins the server.

**Comment by Enchufa2** [comment, marked as answer]
Yes, it is. This is the method to add resources:

```
     add_resource(.env, name, capacity = 1, queue_size = Inf, mon = TRUE,
       preemptive = FALSE, preempt_order = c("fifo", "lifo"),
       queue_size_strict = FALSE, queue_priority = c(0, Inf))
```

As you can see, there's `capacity=1` and an infinite queue by default. You can increase the capacity to be able to serve as many clients at the same time as required. In your case, setting `capacity=4` will do.

**Reply by augustodarruda** [reply]
Hmmmm, i did use capacity = 4 but that does not seems quite right, let me try to explain (sorry, english not native tongue)

Let's say I have 80 servers (80 people that can chat live with a customer). The thing is, each server can live chat simultaneously with 4 customers. I was using:

`call <- 
  simmer("call1") %>% 
  add_resource("server",80) %>%
  add_generator("client", client, function() {c(0,rexp(999,1/2),-1)}, mon = 2)`
  
  But in that way, I assume each server only has the capacity of serving 1 customer only. When i set capacity = 4, it seems like I was telling the code to add only 4 servers, and not that I wanted each server to handle 4 clients at the same time.
  
  Where am I failing to understand?
  
  Thanks for the patience

**Reply by Enchufa2** [reply]
So you want:

1. 80 _different_ resources, each one with a different queue (so 80 queues), and each resource serves up to 4 clients? Then:

```r
call <- simmer("call1") %>%
  add_resource(paste("server", 1:80), 4)
```

2. A single queue for 80 servers that can take up to 4 clients? Then:

```r
call <- simmer("call1") %>%
  add_resource("server", 80 * 4)
```

**Reply by augustodarruda** [reply]
YES. For the time being the second solution is the best, but eventually I will apply the first one. Thank you very much @Enchufa2
